src/Config.hs

```diff
@@ -47,25 +48,40 @@ instance Show Options where
     , "Secret file:    " ++ oSecretFile opts
     , "Command:        " ++ oCmd opts
     , "Arguments:      " ++ (show $ oArgs opts)
-    , "Use TLS:        " ++ (show . not $ oNoConnectTls opts)
-    , "Validate certs: " ++ (show . not $ oNoValidateCerts opts)
-    , "Inherit env:    " ++ (show . not $ oNoInheritEnv opts)
-    , "Debug:          " ++ (show $ oDebug opts)
+    , "Use TLS:        " ++ (show $ oConnectTls opts)
+    , "Validate certs: " ++ (show $ oValidateCerts opts)
+    , "Inherit env:    " ++ (show $ oInheritEnv opts)
     , "Base delay:     " ++ (show . unMilliSeconds $ oRetryBaseDelay opts)
     , "Retry attempts: " ++ (show $ oRetryAttempts opts)
+    , "Log-level:      " ++ (show $ oLogLevel opts)
     ]
 
 -- | Behavior flags that we allow users to set via environment variables.
 -- This type is internal to the workings of this module. It is used as an
 -- intermediate value to get optparse-applicative to play nice with environment
 -- variables as used for behavior flags. All flags are off by default.
 data EnvFlags = EnvFlags
-  { efNoConnectTls :: Bool
-  , efNoValidateCerts :: Bool
-  , efNoInheritEnv :: Bool
-  , efDebug :: Bool
+  { efConnectTls :: Bool
+  , efValidateCerts :: Bool
+  , efInheritEnv :: Bool
   }
 
+-- | LogLevel to run vaultenv under. Under @Error@, which is the default, we
+-- will print error messages in error cases. Examples: Vault gives 404s, our
+-- token is invalid, we don't have permissions for a certain path, Vault is
+-- unavailable.
+--
+-- Under @Info@, we print some additional information related to the config.
+data LogLevel
+  = Error
+  | Info
+  deriving (Eq, Ord, Show)
```

Comment:
Deriving `Ord` does not do the right thing here: 

```
λ> Error > Info
False
```

That's because GHC will just order them alphabetically, I think.
You have to define your own `Ord` instance to get the desired ordering.

Reply:
Agreed on the order switch! We can actually use the derived `Ord` instance. GHC orders them by the order defined in the sum type. With the new changes:

```haskell
data LogLevel
  = Info
  | Error
  deriving (Eq, Ord)
```

We get:

```ghci
*Main Config> Info < Error
True
*Main Config> Info > Error
False
```